### ISSUE TITLE
distribution of UserManual.pdf (and new name?)

### ISSUE THREAD

**Issue by eroux** [gregorio-project/gregorio#181]
I don't really know what would be the best way to distribute the user manual... Maybe https://github.com/gregorio-project/gregorio/releases would be good? What do you think? It's not ideal, but at least we can differentiate according to versions: if we are to make it on the website, we should append the version to the file name (which is not excluded).

Also, I think it should be renamed with a less generic name, I propose GregorioRef.pdf. What do you think?

**Comment by eschwab** [comment]
Having it in releases may be good, then everyone knows what version it is for. Appending the version to the filename is a good idea no matter what. I would also add the version to the title page, perhaps that could be another job for VersionManager.py.

There needs to be a link to the Manual on the website, or at least a reference to find the Manual for your version of gregorio at gregorio/releases. 

GregorioRef.pdf is fine with me.

**Comment by rpspringuel** [comment]
I'm in accord with all of the proposed.

**Comment by eroux** [comment]
Good idea, this could be handled by `VersionManager.py`! Renaming the pdf to get the final filename can be up to the release manager (not a very accurate wording for "person who builds and uploads the release)

**Comment by henryso** [comment]
Why not just have `VersionManager.py` update the (root) `Makefile.am` to put the right version in a Make variable and use that as a define for C and in the Makefile.am for `UserManual.pdf`?  Is that a dumb idea?

**Comment by henryso** [comment]
Actually, we already have it as $(VERSION).  So why not use that in `docs/Makefile.am` ?

**Comment by eroux** [comment]
hmmm so you would use a sed script in the Makefile to change the `.tex` file and put the right version in it? Why not, but as we have to update a few files with `VersionManager.py`, I don't really see why your solution would be more simple?

**Comment by henryso** [comment]
I thought you just wanted to have the PDF with the right filename.

I'm proposing this:
```
diff -r 13354e7e8ca3 doc/Makefile.am
--- a/doc/Makefile.am   Sun Mar 29 16:40:13 2015 +0200
+++ b/doc/Makefile.am   Sun Mar 29 14:57:06 2015 -0400
@@ -1,11 +1,11 @@
 SRCFILES= Command_Index_gregorio.tex  Command_Index_internal.tex  Command_Index_User.tex UserManual.tex
 
-UserManual.pdf: $(SRCFILES)
-       latexmk -pdf -pdflatex='lualatex --shell-escape %O %S' UserManual.tex
+UserManual-$(VERSION).pdf: $(SRCFILES)
+       latexmk -pdf -pdflatex='lualatex --shell-escape %O %S' -jobname=UserManual-$(VERSION) UserManual.tex
 
-doc: UserManual.pdf
+doc: UserManual-$(VERSION).pdf
 
 clean:
        latexmk -quiet -C
 
-EXTRA_DIST = $(SRCFILES) UserManual.pdf
+EXTRA_DIST = $(SRCFILES) UserManual-$(VERSION).pdf
```

**Comment by eroux** [comment]
Unless you talked about suffixing the pdf file with the version? In this case why not, yes, but you cannot change the output filename in TeX, so it would require `mv UserManual.pdf UserManual-3.0.pdf`, but that would mean rebuilding from the beginning and do the three compilations, etc. so I'm affraid it won't be very simple neither...

**Comment by eroux** [comment]
Well, if it works ok then, I thought it was inducing problems, but that might have been with old versions.

**Comment by eschwab** [comment]
Perhaps when a release is built `make dist`, after building the `.pdf`, could rename the `.pdf` to `GregorioRef-3_0_0.pdf`

`VersionManager.py` takes care of the version on the titlepage, `make` handles the filename. Then the release manager does not need to worry about fixing the filename.

**Comment by eroux** [comment]
Sounds good, I updated the [wiki page](https://github.com/gregorio-project/gregorio/wiki/ReleaseActions)

**Comment by eroux** [comment]
I started a few changes on [fixdoc](https://github.com/gregorio-project/gregorio/tree/fixdoc), but I have a problem: with Henry's patch, I get a file with several dots, and it will confuse a few systems. If I want to define a new version number suited for filenames, I need to do it in `configure.ac`, I could:

 * get the version through `VersionManager.py -f` (a new option to get a version suitable for filenames), but this would make the build process depend on python, and I think it's a bad idea
 * get the version by parsing `.gregorio-version`, but it's not really in the spirit of `VersionManager.py`
 * reverse the commit and let the release manager handle things
 * think about another solution, but I don't see any simple one...

What do you think?

**Comment by eschwab** [comment]
Could there be a filename friendly version number in `doc/makefile.am`? Then Henry's idea would still work if it used `$(FILENAME_VERSION)` instead of `$(VERSION)`.

**Comment by henryso** [comment]
I take it FILENAME_VERSION should change dots to underscores or something like that?  How do I get the correct value?  From VersionManager.py?

**Comment by eroux** [comment]
I did it in the fixdoc branch. What took me most of the time was to fix the regexp, they should be much more accurate now...

**Comment by eschwab** [comment]
Addressed in #189
